### PR TITLE
Prototype: Scary Vec2s and Nice Sprites

### PR DESCRIPTION
**Pull request by pushfoo**
**TL;DR:** Proof we ***can*** `Vec2` all `*Sprite` without destroying the universe!

[\[ Obligatory Skrillex Noises \]](https://www.youtube.com/watch?v=WSeNSzJ2-Jw)

>[!NOTE]
>The remaining pyright issues belong in #2183

#### In your Arcade clone

1. `git checkout upstream/scary-vecs-and-nice-sprites` or however you have your remotes instealled
2. `pip install -I -e .[dev]`
   * This is very important
   * Do it every time you switch to or back from this branch
 3. Proceed as normal, see what breaks

#### In your Project

For the latest unstable version:
`arcade@git+https://github.com/pythonarcade/arcade/arcade.git@scary-vecs-and-nice-sprites#egg=arcade`

You can also pin specific commits using a zipball link, but that's more convoluted.

### Changes

1. For `BasicSprite` and subclasses, the following are now `Vec2`:
   * `.size` is now a `Vec2` backed by `._size`
   * `.position` is now a `Vec2`
   * `.scale` is now a `Vec2`
5. `_width` and `_height` are now compatibility properties for any subclasses we missed
6. Dependent functions are mostly updated

### What's not changed

Anything outside the sprites really. Seriously. There's a sound tweak to get it to pass and run, but otherwise, not even tests.

### Current Code Quality
Already done:
- [x] #2226
- [x] #2021

Sprites pass:
- [x] `pytest tests/unit`
- [x] `pyright`
- [x] `./make./py format` + `ruff check arcade`

Otherwise, no guarantees:
image

### Follow-up Work

- [ ] Verify the tests cover what they should 
- [ ] Perf tests and metrics to decide how to proceed with:

### CODE REVIEW

**Comment by pushfoo** [comment]
Right, I need to figure out the zipballs since it's dependent on https://github.com/pyglet/pyglet/pull/1154

**Comment by pushfoo** [comment]
Zipball's fixed, but the other pyright issues belong in #2183.

File: arcade/sprite/base.py [hunk 1]
```diff
@@ -142,14 +172,16 @@ def depth(self, new_value: float):
...
+        return self._size[0]
 
     @width.setter
     def width(self, new_value: float):
-        if new_value != self._width:
-            self._scale = new_value / self._texture.width, self._scale[1]
-            self._hit_box.scale = self._scale
-            self._width = new_value
+        size = self._size
+        if new_value != size[0]:
```
Comment (DragonMoffon) [inline arcade/sprite/base.py]: Just a style pref would go with 
```
if new_value == size[0]:
   return
```

Comment (einarf) [reply arcade/sprite/base.py]: Yup. Definitely do early return instead

**Comment by pushfoo** [comment]
I've dropped the zipball commits during rebase onto `development`. We're on `pyglet==2.1dev4` now.

**Comment by DragonMoffon** [comment]
Maybe set this PR to draft for now?

**Comment by pushfoo** [comment]
@einarf @Cleptomania Given the pyglet 3.0 work, should I close this or leave it here as a reminder to look into the vectors?

**Comment by einarf** [comment]
We can just leave it here for now
